paddle/fluid/operators/conv_cudnn_op.cu

```diff
@@ -68,33 +72,70 @@ class CUDNNConvOpKernel : public framework::OpKernel<T> {
     const std::string data_format = ctx.Attr<std::string>("data_format");
     const bool channel_last = (data_format == "NHWC" || data_format == "NDHWC");
 
+    auto dtype = platform::CudnnDataType<T>::type;
+
+    // Tensor Core introduced from Volta GPUs supports more faster conv op
+    // with FP16 in NHWC data format.
+    const bool compute_in_nhwc =
+        dtype == CUDNN_DATA_HALF && IsVoltaOrLater(dev_ctx);
+    // We will only do data format conversion from NHWC to NCHW.
+    // cudnn will convert NCHW to NHWC automatically on Tensor Core.
+    auto compute_format = compute_in_nhwc && channel_last ? "NHWC" : "NCHW";
+    VLOG(3) << "Compute ConvOp with cuDNN:"
+            << " data_format=" << data_format
+            << " compute_format=" << compute_format;
+
     // ------------ transformed tensor -----------
     Tensor transformed_input_channel(input->type());
     Tensor transformed_output(output->type());
+    Tensor transformed_filter_channel(filter->type());
     T* output_data = nullptr;
-    if (channel_last) {
+    if (channel_last && compute_format == "NCHW") {
+      VLOG(3) << "Transform input tensor from NHWC to NCHW.";
       ResizeToChannelFirst<platform::CUDADeviceContext, T>(
           ctx, input, &transformed_input_channel);
       TransToChannelFirst<platform::CUDADeviceContext, T>(
           ctx, input, &transformed_input_channel);
 
+      ResizeToChannelFirst<platform::CUDADeviceContext, T>(
+          ctx, filter, &transformed_filter_channel);
+      TransToChannelFirst<platform::CUDADeviceContext, T>(
+          ctx, filter, &transformed_filter_channel);
+
       ResizeToChannelFirst<platform::CUDADeviceContext, T>(ctx, output,
                                                            &transformed_output);
 
     } else {
-      transformed_input_channel = *input;
-      transformed_output = *output;
+      transformed_input_channel.ShareDataWith(*input);
+      transformed_output.ShareDataWith(*output);
+    }
+    if (compute_format == "NHWC") {
+      VLOG(3) << "Transform filter tensor from NCHW to NHWC.";
+      ResizeToChannelLast<platform::CUDADeviceContext, T>(
+          ctx, filter, &transformed_filter_channel);
+      TransToChannelLast<platform::CUDADeviceContext, T>(
+          ctx, filter, &transformed_filter_channel);
```

Comment:
is the layout of the filter is consistent with the input? If not so, why not resize the layout of the filter to be the same with the input in the previous `else` block.

Reply:
See last comment